scratch.Rmd

```diff
@@ -0,0 +1,556 @@
+  arrange(ymd_hms(DT)) %>%
+  group_by(site,DT) %>%
+  summarize(Water_Temp_C = mean(Water_Temp_C),
+            PVC_Temp = mean(PVC_Temp)) %>%
+  ungroup() %>%
+  distinct(DT,.keep_all=T) %>%
+  # pass one (p1): remove any data outside the In-Situ spec range
+  mutate(p1 = ifelse(Water_Temp_C > 50, NA,
+              ifelse(Water_Temp_C < 0 , NA, Water_Temp_C)))
+  # mutate(pre = lag(Water_Temp_C, 1),
+  #        post = lead(Water_Temp_C, 1),
+  #        roll = (pre+Water_Temp_C+post)/3)
+
+means <- rollmean(legacy_temp$Water_Temp_C, k = 5, align = "center")
+
+compare <- legacy_temp %>%
+  select(DT, Water_Temp_C, PVC_T, site) %>%
+  #rbind(usgs_lincoln) %>%
+  pivot_wider(., values_from = Water_Temp_C, names_from = "site")
+
+timeseries <- xts::xts(compare, order.by = ymd_hms(compare$DT))
+
+dygraph(timeseries, main="Water Temperature Timeseries (UTC)") %>% 
+ dyRangeSelector() #%>%
+ # dyHighlight(highlightCircleSize = 4,
+ #           highlightSeriesBackgroundAlpha = 0.5,
+ #            hideOnMouseOut = T) %>%
+ # dyLegend(show = "follow")
+```
+
+
+## Legacy depth
+
+We had a couple issues with "level" calibration. Here, I am "back-calibrating" for them to match previous and/or post level when they were correct. Assumes (though subjectively verified in field) that river depth did not change during the field visit.
+```{r}
+legacy_depth <- dplyr::select(rawless_legacy, c(DT, Depth_ft, site)) %>%
+  arrange(ymd_hms(DT)) %>%
+  group_by(site,DT) %>%
+  summarize(Depth_ft = mean((Depth_ft))) %>%
+  ungroup() %>%
+  distinct(DT,.keep_all=T) #%>%
+
+legacy_depth <- legacy_depth %>%
+  mutate(Depth_ft_p1 = ifelse(DT >= ymd_hms('2022-04-06 06:00:00') & DT < ymd_hms('2022-04-12 09:30:00'), Depth_ft + abs(filter(legacy_depth, DT == ymd_hms('2022-04-12 09:15:00'))$Depth_ft - filter(legacy_depth, DT == ymd_hms('2022-04-12 09:30:00'))$Depth_ft), Depth_ft)) %>%
+  mutate(Depth_ft_p1 = ifelse(DT >= ymd_hms('2022-07-22 11:30:00') & DT <= ymd_hms('2022-07-25 14:15:00'), Depth_ft_p1 + abs(filter(legacy_depth, DT == ymd_hms('2022-07-22 11:15:00'))$Depth_ft_p1 - filter(legacy_depth, DT == ymd_hms('2022-07-22 11:30:00'))$Depth_ft_p1), Depth_ft_p1)) %>%
+  mutate(pre = lag(Depth_ft_p1, 1),
+         post = lead(Depth_ft_p1, 1),
+         roll = (pre+Depth_ft_p1+post)/3)
+
+compare <- legacy_depth %>%
+  select(DT, Depth_ft=Depth_ft_p1, site) %>%
+  rbind(usgs_lincoln) %>%
+  pivot_wider(., values_from = Depth_ft, names_from = "site") %>%
+  dplyr::filter(DT >= ymd_hms('2019-09-06 17:30:00'))
+
+timeseries <- xts::xts(compare, order.by = ymd_hms(compare$DT))
+
+dygraph(timeseries, main="Depth Timeseries (UTC)") %>% 
+ dyRangeSelector()
+```
+
+## Legacy conductivity
+```{r}
+legacy_conductivity <- dplyr::select(rawless_legacy, c(DT, Specific_Conductivity_µS_cm, site)) %>%
+  arrange(ymd_hms(DT)) %>%
+  group_by(site,DT) %>%
+  summarize(Specific_Conductivity_µS_cm = mean((Specific_Conductivity_µS_cm))) %>%
```

Comment:
is this needed because of dupe rows? Before you summarize, consider why it is needed and if there is a way you can fix it in your data instead of summarizing. If you *do* need to summarize in this way, your variable is no longer just Spec Cond, it's an average Spec Cond (changing the col name), and it would be helpful for an end user to know how many values contribute to that mean value (`n_SpecCond = length(SpecCond)`), and what the standard deviation was (`SD_SpecCond = sd(SpecCond)`).  IMO, you're creating a 'Level 2' dataset where you are augmenting the observations. While this may be necessary for your uses, another user might not agree.